### PR TITLE
Add comprehensive installation instructions

### LINKED ISSUE #192: Instructions on how to build debian package
sanoid comes with `debian/` directory, but it would be helpful if the README said how to build the package.

After some experimentation, the following appears to work:

~~~
sudo apt-get install devscripts debhelper dh-systemd
git clone https://github.com/jimsalterjrs/sanoid.git
cd sanoid
debuild -us -uc
~~~

although it generates a bunch of lintian errors and warnings.  The .deb package is written in the enclosing directory:

~~~
cd ..
sudo apt-get install libconfig-inifiles-perl
sudo dpkg -i sanoid_1.4.16_all.deb
~~~

### PR DESCRIPTION
**Pull request by RulerOf**
Converts the old INSTALL file to a markdown-formatted INSTALL.md file
that contains comprehensive installation instructions for supported
operating systems, and an outline that can be used on others where
Sanoid/Syncoid ought to work.

The Installation section is complete for CentOS/RHEL, and has been
"dressed up" for FreeBSD/Ubuntu. Those ones still need to be completed.
I don't cover the topic of running Syncoid periodically or how to do it
because I don't know. It's probably out of scope for the Installation
guide anyway.

The Configuration sections is intended to walk through some common
setups for sanoid.conf. I suggest just cleaning up the existing conf
file examples and putting them into that section.

Fixes #192 

----

This is a followup on [this Reddit discussion](https://www.reddit.com/r/zfs/comments/865p34/deleting_snapshots/dw4myck/) from a few days ago.  Let me know if there's anything that should be changed or otherwise cleaned up.

### CODE REVIEW

**Comment by jimsalterjrs** [comment]
So, I like the work you did here, but I'm a little wishy-washy about taking it. Markdown is pretty on Github, but it's more of a PITA to parse on the command line, and I'm partial to the idea of being able to read the install docs with a simple `less` when you're actually on the console and working.

Thoughts?

**Comment by jimsalterjrs** [comment]
I'm definitely okay with this as a separate document designed/intended to be read on Github and linked to from the main readme..

There is the issue of then needing potentially to maintain two different sets of instructions, but then again, it gives us the chance to take some baby steps and perhaps in the future look at who's using which more frequently.

**Comment by backamblock** [comment]
i would really like to use sanoid and syncoid but i am having a hard time installing and configurng it, can you please update this instruction and include it in /master?

**Comment by RulerOf** [comment]
@backamblock When I originally wrote this, I had to leave some parts either blank or unfilled because I didn't have enough information to finish it properly at the time, and had sort of hoped someone else might step in.

I'll carve out some time to spend an hour or two on it either this weekend or next, and will add the instructions necessary for use on Ubuntu/Debian as well.

@phreaker0 if you want to add me as a reviewer, I'll tag this PR with my review once I update it.

**Comment by phreaker0** [comment]
@jimsalterjrs are you still objected to markdown? I would also like to have a single install instruction file, if you are against markdown I would suggest to adapt the plaintext install instruction with the information in this PR (thx @RulerOf)

**Comment by jimsalterjrs** [comment]
Eh, I'm OK with markdown. Honestly it's all sort of baby steps that will/should eventually not matter anyway; one of these days we really need to just have real packages available for the majority distros.

**Comment by phreaker0** [comment]
@jimsalterjrs @RulerOf  I rebased the branch and added recent changes

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+[Unit]
+Description=Snapshot ZFS Pool
+Requires=zfs.target
+After=zfs.target
+ConditionFileNotEmpty=/etc/sanoid/sanoid.conf
+
+[Service]
+Environment=TZ=UTC
+Type=oneshot
+ExecStart=/usr/sbin/sanoid --take-snapshots
```
Comment (danielewood) [inline INSTALL.md]:
```suggestion
ExecStart=/usr/local/sbin/sanoid --take-snapshots
```

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+[Unit]
+Description=Cleanup ZFS Pool
+Requires=zfs.target
+After=zfs.target sanoid.service
+ConditionFileNotEmpty=/etc/sanoid/sanoid.conf
+
+[Service]
+Environment=TZ=UTC
+Type=oneshot
+ExecStart=/usr/sbin/sanoid --prune-snapshots
```
Comment (danielewood) [inline INSTALL.md]:
```suggestion
ExecStart=/usr/local/sbin/sanoid --prune-snapshots
```

**Pull request merged**